### ISSUE TITLE
non intuitive-syntax with supposedly OPTIONAL arguments

### ISSUE THREAD

**Issue by ifenty** [podaac/data-subscriber#39]
I use python, I use the command line.  installed data-subscriber. ran podaac-data-subscriber -h, quickly read the syntax, tried to download all files for a single ECCO dataset into the current directory.  According to the docs, the **-sd STARTDATE** and **-ed ENDATE** are OPTIONAL, but just try to run without them -- failure:

Also failing (not shown) is only using the **-ed** flag to get *ALL* available files from the beginning to the specified end date.

The only combination that works is when both **-sd** and **-ed** are specified.

```
$ podaac-data-subscriber -c ECCO_L4_ATM_STATE_05DEG_DAILY_V4R4 -ed 1993-01-01T00:00:00Z -d newDirectory
```

```
NOTE: Making new data directory at newDirectory(This is the first run.)
Traceback (most recent call last):
  File "/home/ifenty/anaconda3/envs/ecco/bin/podaac-data-subscriber", line 8, in <module>
    sys.exit(run())
  File "/home/ifenty/anaconda3/envs/ecco/lib/python3.8/site-packages/subscriber/podaac_data_subscriber.py", line 345, in run
    with urlopen(url) as f:
  File "/home/ifenty/anaconda3/envs/ecco/lib/python3.8/urllib/request.py", line 222, in urlopen
    return opener.open(url, data, timeout)
  File "/home/ifenty/anaconda3/envs/ecco/lib/python3.8/urllib/request.py", line 531, in open
    response = meth(req, response)
  File "/home/ifenty/anaconda3/envs/ecco/lib/python3.8/urllib/request.py", line 640, in http_response
    response = self.parent.error(
  File "/home/ifenty/anaconda3/envs/ecco/lib/python3.8/urllib/request.py", line 569, in error
    return self._call_chain(*args)
  File "/home/ifenty/anaconda3/envs/ecco/lib/python3.8/urllib/request.py", line 502, in _call_chain
    result = func(*args)
  File "/home/ifenty/anaconda3/envs/ecco/lib/python3.8/urllib/request.py", line 649, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 400: Bad Request
```

**Comment by mike-gangl** [comment]
thanks for the feedback. They are optional in the sense they are not required to run. when specified both must be used in the current implementation. we can update documentation while thinking of a better long term option.

**Comment by ifenty** [comment]
Cool.  If one need to specify both  **-sd** and **-ed** to download anything, then it might be better to call them required.

With the example below, skipping the optional arguments yields zero files downloaded.

```
$ podaac-data-subscriber -c ECCO_L4_ATM_STATE_05DEG_DAILY_V4R4 -d MG_test
NOTE: Making new data directory at MG_test(This is the first run.)
Downloaded: 0 files

Files Failed to download:0

CMR token successfully deleted
```

**Comment by mike-gangl** [comment]
i totally agree. I think this is the difference (in theory) between a subscriber and a bulk downloader. the subscriber initial writing was to run often, and gather data since the last time it run. The default run only look back 60 minutes for new data. Perhaps that's an erroneous assumption, and the -sd/-ed options should have always been required.

**Comment by mike-gangl** [comment]
i was mistaken, we actually had a bug that affected the case where only -ed was give. had only -sd been given, it would have been fine. This will be fixed in the 1.8 version of the subscriber. we'll also be introducing a data-downloader that is probably better for your use case (as opposed to continuously pulling data as it comes in via the subscriber)

the data-downloader will require both start/end dates to work.

**Comment by mike-gangl** [comment]
```
python subscriber/podaac_data_subscriber.py -c ECCO_L4_ATM_STATE_05DEG_DAILY_V4R4 -ed 1993-01-01T00:00:00Z -d newDirectory
NOTE: Making new data directory at newDirectory(This is the first run.)
Found 367 total files to download
2022-02-01 23:08:14.862596 SUCCESS: https://archive.podaac.earthdata.nasa.gov/podaac-ops-cumulus-protected/ECCO_L4_ATM_STATE_05DEG_DAILY_V4R4/ATM_SURFACE_TEMP_HUM_WIND_PRES_day_mean_1993-01-01_ECCO_V4r4_latlon_0p50deg.nc
2022-02-01 23:08:16.198288 SUCCESS: https://archive.podaac.earthdata.nasa.gov/podaac-ops-cumulus-protected/ECCO_L4_ATM_STATE_05DEG_DAILY_V4R4/ATM_SURFACE_TEMP_HUM_WIND_PRES_day_mean_1992-12-31_ECCO_V4r4_latlon_0p50deg.nc
2022-02-01 23:08:17.524378 SUCCESS: https://archive.podaac.earthdata.nasa.gov/podaac-ops-cumulus-protected/ECCO_L4_ATM_STATE_05DEG_DAILY_V4R4/ATM_SURFACE_TEMP_HUM_WIND_PRES_day_mean_1992-12-30_ECCO_V4r4_latlon_0p50deg.nc
2022-02-01 23:08:18.858536 SUCCESS: https://archive.podaac.earthdata.nasa.gov/podaac-ops-cumulus-protected/ECCO_L4_ATM_STATE_05DEG_DAILY_V4R4/ATM_SURFACE_TEMP_HUM_WIND_PRES_day_mean_1992-12-29_ECCO_V4r4_latlon_0p50deg.nc
2022-02-01 23:08:20.360686 SUCCESS: https://archive.podaac.earthdata.nasa.gov/podaac-ops-cumulus-protected/ECCO_L4_ATM_STATE_05DEG_DAILY_V4R4/ATM_SURFACE_TEMP_HUM_WIND_PRES_day_mean_1992-12-28_ECCO_V4r4_latlon_0p50deg.nc
```
